src/com/mendix/widget/badge/components/Badge.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, DOM, createElement } from "react";
+import * as classNames from "classnames";
+
+export type BadgeOnclick = "doNothing" | "showPage" | "callMicroflow";
+export type PageSettings = "content" | "popup" | "modal";
+
+export interface MicroflowProps {
+    microflow: string;
+    guid: string;
+}
+
+export interface PageProps {
+    page: string;
+    pageSetting: PageSettings;
+    entity: string;
+    guid: string;
+}
+
+export interface OnClickProps {
+    onClickType: BadgeOnclick;
+    microflowProps?: MicroflowProps;
+    pageProps?: PageProps;
+}
+
+export interface BadgeProps {
+    label?: string;
+    badgeType: string;
+    badgeValue?: string;
+    style?: string;
+    badgeOnClick?: OnClickProps;
+    disabled?: string;
+}
+
+export class Badge extends Component<BadgeProps, {}> {
+    static defaultProps: BadgeProps = {
+        badgeType: "badge",
+        badgeValue: "0",
+        label: "default",
+        style: "default"
+    };
+    componentDidMount() {
+        this.checkConfig();
+        this.setBadge();
+    }
+
+    componentDidUpdate() {
+        this.setBadge();
+    }
+
+    render() {
+        return this.setBadge();
+    }
+
+    private setBadge() {
```

Comment:
Name of function does not represent the content... 
It's is used only once (see other comments) so it could be inlined in the `render`